From f0971cf18afbef93273a60095965f6f9eebe799f Mon Sep 17 00:00:00 2001
From: Kalyan Kondapally <kalyan.kondapally@intel.com>
Date: Sun, 20 Mar 2016 02:12:31 -0700
Subject: [PATCH 3/3] Query initial bound fb from RenderContext in
 FrameBufferBindingTestCase

Initial bound FB might differ based on surface type i.e. It's 1
when --deqp-surface-type=fbo. Instead of assuming 0 always,
query it from RenderContext.

Test: Check that the following tests pass:
1)dEQP-GLES2.functional.state_query.integers.framebuffer_binding_getboolean
2)dEQP-GLES2.functional.state_query.integers.framebuffer_binding_getfloat
3)dEQP-GLES2.functional.state_query.integers.framebuffer_binding_getinteger
4)dEQP-GLES2.functional.state_query.integers.framebuffer_binding_getboolean
5)dEQP-GLES2.functional.state_query.integers.framebuffer_binding_getfloat
6)dEQP-GLES2.functional.state_query.integers.framebuffer_binding_getinteger
7)dEQP-GLES2.functional.state_query.integers.framebuffer_binding_getinteger64
---
 modules/gles2/functional/es2fIntegerStateQueryTests.cpp | 4 +++-
 modules/gles3/functional/es3fIntegerStateQueryTests.cpp | 8 +++++---
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/modules/gles2/functional/es2fIntegerStateQueryTests.cpp b/modules/gles2/functional/es2fIntegerStateQueryTests.cpp
index 9e875a1..3ace9ac 100644
--- a/modules/gles2/functional/es2fIntegerStateQueryTests.cpp
+++ b/modules/gles2/functional/es2fIntegerStateQueryTests.cpp
@@ -1859,7 +1859,9 @@ public:
 
 	void test (glu::CallLogWrapper& gl, tcu::ResultCollector& result) const
 	{
-		verifyStateInteger(result, gl, GL_FRAMEBUFFER_BINDING, 0, m_type);
+                // The initial bound FB can differ based on surface type i.e. It's 1 when --deqp-surface-type=fbo.
+                // Query it from RenderContext.
+		verifyStateInteger(result, gl, GL_FRAMEBUFFER_BINDING, m_context.getRenderContext().getDefaultFramebuffer(), m_type);
 
 		GLuint framebufferId = 0;
 		gl.glGenFramebuffers(1, &framebufferId);
diff --git a/modules/gles3/functional/es3fIntegerStateQueryTests.cpp b/modules/gles3/functional/es3fIntegerStateQueryTests.cpp
index 7760d11..81678b8 100644
--- a/modules/gles3/functional/es3fIntegerStateQueryTests.cpp
+++ b/modules/gles3/functional/es3fIntegerStateQueryTests.cpp
@@ -2442,9 +2442,11 @@ public:
 
 	void test (glu::CallLogWrapper& gl, tcu::ResultCollector& result) const
 	{
-		verifyStateInteger(result, gl, GL_DRAW_FRAMEBUFFER_BINDING,	0, m_type);
-		verifyStateInteger(result, gl, GL_FRAMEBUFFER_BINDING,		0, m_type);
-		verifyStateInteger(result, gl, GL_READ_FRAMEBUFFER_BINDING,	0, m_type);
+               // The initial bound FB can differ based on surface type i.e. It's 1 when --deqp-surface-type=fbo.
+               // Query it from RenderContext.
+                verifyStateInteger(result, gl, GL_DRAW_FRAMEBUFFER_BINDING,	m_context.getRenderContext().getDefaultFramebuffer(), m_type);
+                verifyStateInteger(result, gl, GL_FRAMEBUFFER_BINDING,		m_context.getRenderContext().getDefaultFramebuffer(), m_type);
+                verifyStateInteger(result, gl, GL_READ_FRAMEBUFFER_BINDING,	m_context.getRenderContext().getDefaultFramebuffer(), m_type);
 
 		GLuint framebufferId = 0;
 		gl.glGenFramebuffers(1, &framebufferId);
-- 
2.5.0

